hangman.py

```diff
@@ -0,0 +1,240 @@
+    if count_star != 0:
+        warnings_point, guesses_point = warnings("you are playing classic hangman", warnings_point, guesses_point)
+    else:
+        print("Oops, you are playing classic hangman:", end=" ")
+        count_star += 1
+    return warnings_point, guesses_point, count_star
+
+
+def hangman(secret_word, request):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+
+    count_star is variable, which count all input star. It need if user play in hangman with hints and input star more
+    than 30 times, program output random possible words.
+    '''
+
+    count_star = 0
+    guesses_point = 6
+    warnings_point = 3
+    letters_guessed = set()
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", end=" ")
+        get_available_letters(letters_guessed)
+        letter = input("Please guess a letter:")
+        if request is False and letter == STAR:
+            check_star_30(count_star, secret_word, letters_guessed)
+            count_star += 1
+        else:
+            if letter == STAR:
+                warnings_point, guesses_point, count_star = classic_input(count_star, warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter not in string.ascii_letters or len(letter) != 1:
+                warnings_point, guesses_point = warnings("That is not a valid letter or you print more than one letter.", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter in letters_guessed:
+                warnings_point, guesses_point = warnings("You have already guessed that letter:", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            else:
+                letters_guessed, guesses_point = letter_in_word(secret_word, letter, letters_guessed, guesses_point, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0 or guesses_point <= 0:
+            break
+
+    #output result of game
+    if is_word_guessed(secret_word, letters_guessed) == 0:
+        final_points = guesses_point * len(set(secret_word))
+        print("------")
+        print("Congratulations, you won!")
+        print("Word:", secret_word)
+        print("Your total score for this game is:", final_points)
+    elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+        print("------")
+        print("Sorry, you ran out of guessed. The word was", secret_word)
```

Comment:
No that function is quite long, could you try to achieve it to be up to 30 lines long?